### DISCUSSION TITLE
Updates to Install Instructions - DUI Browser ERRORs

### DISCUSSION THREAD

**Discussion by lvto2000** [kibook/pmms#88] [Ideas]
**TO THE PMMS DEVELOPER(S)**

you have done a good job of creating an app that is very useful in helping developers quickly integrate multimedia into FiveM servers.  However for newcomers to your app, your install or troubleshooting instructions are in need of some updates even for very seasoned programmers.   Whether you choose to do it or not is up to you.  

Still, I am documenting my experience here so that others who run into the same issues I did, will not have to dig into your issues pages to piece together the problem.  Were it not for the fact that I had this problem years ago when I first used your program, I would have given up assuming that your program was buggy and not well maintained.  Here I am returning years later and the same issue still has not been documented.  

As an example and based on the resolution in  issue https://github.com/kibook/pmms/issues/56 dated in 2022, there seems to be long standing challenges with using ZIP instead of GIT to successfully download all the files needed to execute install and use of your program. If you are aware that some users opt for using ZIP (a standard feature in GitHub) instead of GIT to download your application, then why not include that warning in your install instructions instead of forcing people who do use that option to spend hours troubleshooting a known issue?

**SOME TROUBLE SHOOTING HISTORY and OBSCURE PREREQUISITES **

This issue relates to the dreaded DUI error that looks like the image below or you received the INVALID FILE TYPE errors as a result of directly entering the https or http url

![image](https://github.com/kibook/pmms/assets/9310912/6acd851d-383d-4bda-a178-0b09df9d5b4a)

All of the above may be related to the same issue I experienced where I downloaded the core pmms application using ZIP.   In that case,  it means you did not download the helper pmms-dui submodule that is needed.   

If so, then download the module here:  https://github.com/kibook/pmms-dui.   (assuming you are continuing to use ZIP)

Unzip the file.  Once you do  then move the file to wherever you installed the core pmms module under its directory called http/dui.  That means unzip the pmms-dui-master.zip file then go to that directory and move all of its contents under the dui directory.  OR, just remove or move the current dui directory then unzip the pmms-dui-master.zip file under the pmms/http directory.  Then move  or change the name of the pmss-dui-master directory to dui.  Beware that the zip file may unzip to multiple directories with the name pmms-dui.  Make sure you are using the directory with the core pmms-dui-master  submodule and html files directly under it. 

Unfortunately, your problems do not stop there.  If you are reading this then you probably were like me and did not see any comprehensive description of exactly where the media files should be uploaded to your FiveM server.  This is especially true  if you just want to serve music or videos directly from the FiveM server directory structure vs set up a separate html media server or use YouTube.    The install instructions and issue tickets can be misleading making you believe that you have a direct interest in what httpmanager is doing or its file system.  This might very well be useful in the future as a part of any advanced capabilities you want to implement but not as a basic install to test or serve media files.  

Additionally,  the file structure within pmms-dui-master you've just downloaded with my help may further blur the issue.  There are other files that would make you belive that moving your media files under this directory make sense to fix this issue.  To make matters worse, the core pmms file structure has a directly called media in it that seems very tantalizing to move your media files for serving up content.  None of these worked for me while troubleshooting this problem.  As a matter of fact, there is even guidance to pay attention to these file systems as a workaround to the use of https vs. http urls for security reasons and to overcome the same types of errors I am describing here.  As of February 26, 2024, I have found that NONE of this guidance was useful in getting just a rudimentary grasp of getting files served properly.  

**GETTING DOWN TO THE INSTALLING BUSINESS**
If you have just installed all the helper modules I have described above then, as confusing as it sounds, your media files **SHOULD NOT** uploaded to any of these numerous other directory structures.  Strangely, you will have to upload any media files you want to serve in FiveM to the core pmms module subdirectory called stream or pmms/stream.  So, for clarity, once you have installed pmms, pmms-dui-master, and httpmanager dependency modules to your system, move any media files to the pmms module under the pmms/stream.  

After that, follow all the instructions for providing the right permissions (permissions.cfg) and startup order (in server.cfg) to allow the right user types to start and execute video/music.  You might even have to restart your fivem server to get everything in sync.  Once you have everything in place then all you have to do is open pmms by executing the command in-game via the console or chat (/pmms).  

Once the interface is up then just enter the name of the file for the video or music media file you uploaded to the pmms/stream directory exactly as it is spelled and all upper and lower case letters with the file extension name (*.mp3 or *.mp4).  Make sure you select the closest media playing device then hit play.  The local file should play.  Give it about 3-5 seconds to start.

**Comment by kibook** [comment]
Thanks for writing this up, I hope it helps anyone still looking to use this resource. 🙂

Most things I made have been for my own use first and foremost, and I mostly uploaded here just to share, and these projects were mainly targeted at other devs (hence why I only put them on my GitHub and didn't post on something like the FiveM forums).

I do want to comment on one thing that I'm curious about:

> To make matters worse, the core pmms file structure has a directly called media in it that seems very tantalizing to move your media files for serving up content.

That is in fact exactly what the "media" folder was placed there to be, and it did work for me when I was active on FiveM. It's possible something may have changed since then, but the URLs for media placed in that folder would be along the lines of: `http://<server IP>:<server port>/pmms/media/myvideo.mp4`.

**Comment by MaverickMaigre78** [comment]
Hi guys. My pmms console dont show up de Imagens of control how can I get a solution for this
